companyweb_base/models/res_partner.py

```diff
@@ -0,0 +1,532 @@
+# Copyright 2021 ACSONE SA/NV
+import logging
+from datetime import datetime
+from hashlib import sha1
+
+import zeep
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+# Companyweb is ok with those key being visible on github
+SERVICE_INTEGRATOR_ID = "acsone"
+SERVICE_INTEGRATOR_SECRET = "ECAB8ACF-9AE1-4E90-BD0D-05A1F47A3FE9"
+
+
+class CompanywebPartner(models.Model):
+    _inherit = "res.partner"
+    cweb_currency_id = fields.Many2one(
+        "res.currency", string="Companyweb currency", readonly=True
+    )
+    cweb_lastupdate = fields.Datetime("Companyweb Last Update", readonly=True)
+    cweb_name = fields.Char("Companyweb Name", readonly=True)
+    cweb_name_enable = fields.Boolean("Companyweb Name Enabled", readonly=True)
+    cweb_jur_form = fields.Char("Companyweb Juridical Form", readonly=True)
+    cweb_jur_form_enable = fields.Boolean(
+        "Companyweb Juridical Form Enabled", readonly=True
+    )
+    cweb_companystatus = fields.Char("Companyweb Company Status", readonly=True)
+    cweb_companystatus_code = fields.Char("Companyweb Company StatusCode", readonly=True)
+    cweb_companystatus_enable = fields.Boolean(
+        "Companyweb Company Status Enabled", readonly=True
+    )
+    cweb_street = fields.Char("Companyweb Street", readonly=True)
+    cweb_street_visible = fields.Boolean("Companyweb Street Visible", readonly=True)
```

Comment:
For Char fields we should not need this `visible` flag, as we can test for emptyness.